analytical_engine/core/fragment/dynamic_fragment_reporter.h

```diff
@@ -374,6 +376,331 @@ class DynamicGraphReporter : public grape::Communicator {
   static const int batch_num_ = 100;
   folly::json::serialization_opts json_opts_;
 };
+
+template <typename T>
+T convert_oid(folly::dynamic node) {}
+
+template <>
+int64_t convert_oid(folly::dynamic node) {
+  return node.asInt();
+}
+
+template <>
+std::string convert_oid(folly::dynamic node) {
+  return node.asString();
+}
+
+template <typename FRAG_T>
+class ImmutableGraphReporter {};
+
+template <typename OID_T, typename VID_T>
+class ImmutableGraphReporter<vineyard::ArrowFragment<OID_T, VID_T>>
```

Comment:
and take care of the naming.
In our repo, ImmutableFragment and ArrowFragment are represent two things. 
Make sure to use the correct one.

Reply:
rename to `ArrowFragmentReporter` and `DynamicFragmentReporter`